src/client_side.cc

```diff
@@ -3304,10 +3307,10 @@ ConnStateData::startPeekAndSplice()
     bio->hold(true);
 
     // Here squid should have all of the client hello message so the
-    // tlsAttemptHandshake() should return false;
+    // tlsAttemptHandshake() should return 0.
     // This block exist only to force openSSL parse client hello and detect
     // ERR_SECURE_ACCEPT_FAIL error, which should be checked and splice if required.
-    if (!tlsAttemptHandshake(this, nullptr)) {
+    if (tlsAttemptHandshake(this, nullptr) < 0) {
```

Comment:
According to the source code comment, `squid should have all of the client hello message`. Thus, zero (i.e., `need more data`) is not a good/expected tlsAttemptHandshake() outcome here. I do not know whether the comment or this condition is wrong, but the two do not match.

Reply:
Indeed whet I thought earlier. However, if that were true the changed API would have worked.

I hindsight this call / attempt is expected to result in either a WANT_READ or WANT_WRITE occurance (ie. wanting the server Hello data). Which is 0, not 1.

So the expected result is non-true / 0 / "fail" to continue with the FwdState call, versus -1 / error / reject to be tunneled. It should never be true / 1 / success here because the handshake is incomplete.

Reply:
Your response does not match the mismatch problem I described. You are saying that the code works correctly. I have not said that the code does not work correctly.

Reply:
You said the comment and code condition do no match. Due to what the code is doing the comment accurately describes the input state, why the call is being done and expected output. So they do match IMO - they are also the same as previously existing before PR 81 changes.

So what wrongness do you see in either the comment or the condition?

Reply:
I have already answered your question in my original comment. This mismatch is not important enough for me to continue this discussion -- if you do not see the mismatch I have described, then it will stay in the code.

Reply:
All I see in your original comment is a statement, then jumping to an incorrect conclusion.

Your second comment then rejects my reply on grounds that I don't believe you outright and provide a solution to the non-existent problem, which you wrongly concluded as existing. Instead I provide the logic by which your conclusion is proven false and the problem non-existent.

Reply:
> If there is something my logic missed out that your hidden logic used, I'd like to know.

Given the relative unimportance of this particular change request, and the significant complexity of its environment, I am (still) not interested in continuing this conversation. I am also content with you thinking that you have proven me wrong. This change request should be considered addressed.